eng/scripts/Language-Settings.ps1

```diff
@@ -42,13 +68,169 @@ function Get-AllPackageInfoFromRepo($serviceDirectory)
     $pkgProp.SdkType = $sdkType
     $pkgProp.IsNewSdk = ($isNewSdk -eq 'true')
     $pkgProp.ArtifactName = $pkgName
+    $pkgProp.IncludedForValidation = $false
+    $pkgProp.DirectoryPath = ($pkgProp.DirectoryPath)
+
+    if ($pkgProp.Name -in $DependencyCalculationPackages -and $env:DISCOVER_DEPENDENT_PACKAGES) {
+      Write-Host "Calculating dependencies for $($pkgProp.Name)"
+      $outputFilePath = Join-Path $RepoRoot "$($pkgProp.Name)_dependencylist.txt"
+      $buildOutputPath = Join-Path $RepoRoot "$($pkgProp.Name)_dependencylistoutput.txt"
+
+      if (!(Test-Path $outputFilePath)) {
+        Write-Host "        dotnet build /t:ProjectDependsOn ./eng/service.proj
+          /p:TestDependsOnDependency=`"$($pkgProp.Name)`"
+          /p:IncludeSrc=false /p:IncludeStress=false /p:IncludeSamples=false
+          /p:IncludePerf=false /p:RunApiCompat=false
+          /p:InheritDocEnabled=false /p:BuildProjectReferences=false
+          /p:OutputProjectFilePath=`"$outputFilePath`" > $buildOutputPath 2>&1"
+
+        # calculate the dependent packages
+        dotnet build /t:ProjectDependsOn ./eng/service.proj `
+          /p:TestDependsOnDependency="$($pkgProp.Name)" `
+          /p:IncludeSrc=false /p:IncludeStress=false /p:IncludeSamples=false  `
+          /p:IncludePerf=false /p:RunApiCompat=false `
+          /p:InheritDocEnabled=false /p:BuildProjectReferences=false `
+          /p:OutputProjectFilePath="$outputFilePath" > $buildOutputPath 2>&1
```

Comment:
You should be able to replace the separate Write-Host and invoke calls with Invoke-LoggedCommand.